src/lib/y2partitioner/widgets/menus/add.rb

```diff
@@ -54,26 +58,62 @@ def items
 
         # @see Device
         def disabled_for_device
-          disabled = []
-          if !device.is?(:disk_device, :software_raid, :bcache, :partition)
-            disabled << :menu_add_partition
-          end
-          disabled << :menu_add_lv unless device.is?(:lvm_vg, :lvm_lv)
-          disabled
+          items = []
+          items << :menu_add_partition unless support_add_partition?
+          items << :menu_add_lv unless support_add_lv?
+          items
         end
 
         private
 
         # @see Base
+        # rubocop:disable Metrics/CyclomaticComplexity
+        # Is this a complex method actually?
```

Comment:
Well, it likely could be substituted by a hash and/or a little bit of metaprogramming (generating the name of the method and then using `send`).

Reply:
Yes, sure. In fact, I did it with metaprogramming, but the result is more difficult to read/understand. In this kind of things, I am not able to understand why a simple `case` has high complexity.

Reply:
Well, if you tried and it looked worse, then I guess this is fine. I didn't actually tried, I was just assuming it could look good. But experience beats assumptions. :-)

Reply:
well, in general case is usually hidden class separation. Here it is done for actions. So if all actions has simple format of class.new or class.new(device). you can use something like `MENU_ADD_MAPPING[event].new(device)` So hash based factory.

Reply:
and btw if item is proper object, code can look like `items.find { |i| i.id == event }.action` and also above it can create list of items from object like `@items ||= [add_md.item, add_vg.item, ..]`